e2xgrader/apps/api.py

```diff
@@ -140,3 +144,187 @@ def get_task_submissions(self, assignment_id, notebook_id, task_id):
+        source_directory = {}
+        for idx, i in enumerate(nb.cells):
+            if str(nb.cells[idx]['metadata']['nbgrader']['grade']) == 'True'\
+            and str(nb.cells[idx]['metadata']['nbgrader']['solution']) == 'False':
+                source_directory[str(nb.cells[idx]['metadata']['nbgrader']['grade_id'])] = nb.cells[idx]['source']
+        
+        updated_notebook = self.find_notebook(notebook, assignment)
+        source_cells = updated_notebook.source_cells
+        grade_cells = updated_notebook.grade_cells
+
+        cells = [cell.name for cell in grade_cells if cell.cell_type == "code"]
+        autograde_cells = [cell for cell in source_cells if cell.name in cells and cell.locked == True]
+
+        updated_cells = []
+        for cell in autograde_cells:
+            if cell.name in source_directory.keys() and cell.source != source_directory[cell.name]:
+                updated_cells.append(cell.name)
+
+        return updated_cells
+
+    def update_cell_content(self, cell_id: str, notebook: str, assignment: str) -> str:
+        """Updates cell content.
+
+        Arguments
+        ---------
+        cell_id: string
+            The name of the cell
+        notebook: string
+            The name of the notebook
+        assignment: string
+            The name of the assignment
+        cell_content: string
+            The updated content of the cell
+        Returns
+        -------
+        checksum_id: str
+            Generates new checksum id after changes.
+        """
+        nb = nbformat.read("source/" + assignment + "/" + notebook + ".ipynb", as_version = 4)
+        for i in nb.cells:
+            if i['metadata']['nbgrader']['grade_id'] == cell_id:
+                cell_content = i['source']
+                self.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, source = cell_content)
+                self.update_or_create_source_cell(name = cell_id, notebook = notebook, assignment = assignment, checksum = utils.compute_checksum(i))
+
+                return utils.compute_checksum(i)
+
+        return None
+
+    def update_cell(self, cell_id: str, notebook: str, assignment: str) -> str:
+        """Updates cell content.
+
+        Arguments
+        ---------
+        cell_id: string
+            The name of the cell
+        notebook: string
+            The name of the notebook
+        assignment: string
+            The name of the assignment
+        cell_content: string
+            The updated content of the cell
+        Returns
+        -------
+        checksum_id: str
+            Generates new checksum id after changes.
+        """
+        nb = nbformat.read("source/" + assignment + "/" + notebook + ".ipynb", as_version = 4)
+        for i in nb.cells:
+            if i['metadata']['nbgrader']['grade_id'] == cell_id:
+                i['source'] = cell_content
+                fname = "source/" + assignment + "/" + notebook + ".ipynb"
```

Comment:
Use coursedir here to determine source directory and os.path.join to join the path

Reply:
This function was not required and hence deleted.